hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java

```diff
@@ -175,7 +176,11 @@ public void initialize(NodeEngineImpl nodeEngine,
         this.ptionArrgmt = new PartitionArrangement(partitionAssignment, nodeEngine.getThisAddress());
         Set<Integer> higherPriorityVertices = VertexDef.getHigherPriorityVertices(vertices);
         for (Address destAddr : remoteMembers.get()) {
-            memberConnections.put(destAddr, getMemberConnection(nodeEngine, destAddr));
+            Connection conn = getMemberConnection(nodeEngine, destAddr);
+            if (conn == null) {
+                throw new TopologyChangedException("no connection to job participant: " + destAddr);
```

Comment:
Is this hard to reproduce? A test would be great.

Reply:
It's quite hard. The connection to the other member has to be already down, but a moments before when we're building the list of participants it still has to be there. There's a very small window, we could test it only with a test hook that will allow us to wait a while at a specific place.